pyi.py

```diff
@@ -818,6 +823,24 @@ def _is_enum_class(node: ast.ClassDef) -> bool:
     return any(_is_enum_base(base) for base in node.bases)
 
 
+_COMMON_METACLASSES = {"type": "builtins", "ABCMeta": "abc", "EnumMeta": "enum"}
+
+
+def _is_metaclass(node: ast.ClassDef) -> bool:
+    """Best-effort attempt to determine if a class is a metaclass or not."""
+    for base in node.bases:
+        if isinstance(base, ast.Name):
+            if base.id in _COMMON_METACLASSES:
+                return True
+        elif isinstance(base, ast.Attribute):
+            if base.attr in _COMMON_METACLASSES and _is_name(
+                base.value, _COMMON_METACLASSES[base.attr]
+            ):
+                return True
+    else:
+        return False
```

Comment:
We could also cram this into a single expression if we wanted to:

```suggestion
    return any(
        (isinstance(base, ast.Name) and base.id in _COMMON_METACLASSES)
        or (
            instance(base, ast.Attribute)
            and base.attr in _COMMON_METACLASSES
            and _is_name(base.value, _COMMON_METACLASSES[base.attr])
        )
        for base in node.bases
    )
```